### PR TITLE
Scheduler - Virtual Scrolling - Appointments filtering

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

**Review by andrewmakarov — DISMISSED** [review]

**Review by Yarovaya — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js [hunk 1]
```diff
@@ -1610,4 +1610,250 @@ QUnit.module('Integration: Appointments on vertical views (day, week, workWeek)'
         const recurrentAppointmentHeight = scheduler.appointments.getAppointmentHeight(1);
         assert.equal(initialAppointmentHeight, recurrentAppointmentHeight, 'Appointment cells have equal heights (both initial and recurrent)');
     });
+
+    QUnit.module('Virtual Scrolling', function() {
+        ['vertical', 'horizontal'].forEach(groupOrientation => {
+            QUnit.test(`Appointments should filter correctly if scrolling.mode = "virtual", groupOrientation: ${groupOrientation}`, function(assert) {
```
Comment (AryamnovEugeniy) [inline testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js]: ''Appointments should be filtered correctly if ...'

File: testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js [hunk 2]
```diff
@@ -1610,4 +1610,250 @@ QUnit.module('Integration: Appointments on vertical views (day, week, workWeek)'
...
+
+                const filteredItems = this.instance.getFilteredItems();
+
+                assert.equal(filteredItems.length, 3, 'Filtered items length is correct');
+                assert.deepEqual(filteredItems[0], data[0], 'Filtered item 0 is correct');
+                assert.deepEqual(filteredItems[1], data[1], 'Filtered item 1 is correct');
+                assert.deepEqual(filteredItems[2], data[2], 'Filtered item 2 is correct');
+            });
+
+            QUnit.test(`Appointments should filter correctly if resources and scrolling.mode = "virtual", groupOrientation: ${groupOrientation}`, function(assert) {
```
Comment (AryamnovEugeniy) [inline testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js]: Same here

File: testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js [hunk 3]
```diff
@@ -1610,4 +1610,250 @@ QUnit.module('Integration: Appointments on vertical views (day, week, workWeek)'
...
+                const filteredItems = this.instance.getFilteredItems();
+
+                assert.equal(filteredItems.length, 3, 'Filtered items length is correct');
+                assert.deepEqual(filteredItems[0], data[0], 'Filtered item 0 is correct');
+                assert.deepEqual(filteredItems[1], data[1], 'Filtered item 1 is correct');
+                assert.deepEqual(filteredItems[2], data[2], 'Filtered item 2 is correct');
+            });
+        });
+
+        QUnit.test('Appointments should filter correctly if groups, resources and scrolling.mode = "virtual", groupOrientation: "vertical"', function(assert) {
```
Comment (AryamnovEugeniy) [inline testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js]: 'Appointments should be filtered correctly when they are grouped vertically and scrolling.mode = "virtual"'

File: testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js [hunk 4]
```diff
@@ -1610,4 +1610,250 @@ QUnit.module('Integration: Appointments on vertical views (day, week, workWeek)'
...
+            });
+
+            const filteredItems = this.instance.getFilteredItems();
+
+            assert.equal(filteredItems.length, 2, 'Filtered items length is correct');
+            assert.deepEqual(filteredItems[0], data[0], 'Filtered item 0 is correct');
+            assert.deepEqual(filteredItems[1], data[2], 'Filtered item 1 is correct');
+        });
+
+        QUnit.test('Appointments should filter correctly if groups, resources and scrolling.mode = "virtual", groupOrientation: "horizontal"', function(assert) {
```
Comment (AryamnovEugeniy) [inline testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js]: 'Appointments should be filtered correctly when they are grouped horizontally and scrolling.mode = "virtual"'

**Review by AryamnovEugeniy — DISMISSED** [review]

**Review by AryamnovEugeniy — APPROVED** [review]

**Pull request merged**